hardware/Tellstick.cpp

```diff
@@ -238,8 +238,7 @@ bool CTellstick::StopHardware()
     m_bIsStarted = false;
     m_cond.notify_all();
     lock.unlock();
-    if (m_thread.joinable()) //TODO: Why do we only join the thread if it's joinable?
-        m_thread.join();
+    m_thread.join();
```

Comment:
Everywhere in the code we perform

```
if (m_thread)
{
 m_thread,join();
}
```

Of course it is safe (now....!) to assume there is always a thread.. if StartHardware is called... but who knows in the feature...

Reply:
and a m_thread.reset

Reply:
OK, fixed!